utils/openshift_resource.py

```diff
@@ -28,6 +28,48 @@ def __init__(self, body, integration, integration_version,
         self.error_details = error_details
         self.verify_valid_k8s_object()
 
+    def __eq__(self, other):
+        return self.obj_intersect_equal(self.body, other.body)
+
+    def obj_intersect_equal(self, obj1, obj2):
+        if obj1.__class__ != obj2.__class__:
+            return False
+        elif isinstance(obj1, dict):
+            for obj1_k, obj1_v in obj1.items():
+                obj2_v = obj2.get(obj1_k, None)
+                if not obj2_v:
```

Comment:
`if obj2_v is not None` (so you don't enter the `if` when `obj2` is `0` or `''` )

Reply:
thanks!
look at what i did. does it makes sense?

Reply:
yep, looks great. very easy to read.